testnet/stacks-node/src/run_loop/puppet.rs

```diff
@@ -0,0 +1,176 @@
+                                                    puppet_control.block_interval =
+                                                        Duration::from_secs(v);
+                                                    puppet_control.next_block_time =
+                                                        SystemTime::now()
+                                                            .add(puppet_control.block_interval);
+                                                }
+                                            }
+                                            _ => (),
+                                        }
+                                        Ok(Response::new(StatusCode::Ok))
+                                    }
+                                    (Method::Put, "/until") => {
+                                        let body = req.body_string().await;
+                                        match body {
+                                            Ok(x) => {
+                                                let v = x.parse::<u64>().unwrap_or(0);
+                                                if v > 0 {
+                                                    println!("Setting target block to {}", v);
+                                                    io::stdout().flush().unwrap();
+                                                    let mut puppet_control =
+                                                        puppet_control.lock().unwrap();
+                                                    puppet_control.target_block = v;
+                                                }
+                                            }
+                                            _ => (),
+                                        }
+                                        Ok(Response::new(StatusCode::Ok))
+                                    }
+                                    _ => {
+                                        let mut rs = Response::new(StatusCode::BadRequest);
+                                        rs.set_body(format!(
+                                            "[{}] {}",
+                                            req.method(),
+                                            req.url().path()
+                                        ));
+                                        Ok(rs)
+                                    }
+                                }
+                            })
+                            .await
+                            .unwrap_or(())
+                        }
+                        Ok(())
+                    })
+                })
+                .unwrap(),
+        )
+    }
+
+    pub fn block_on(&self, chain_tip: &ChainTip) {
+        if self.join_handle.is_none() {
+            return;
+        }
+        info!(
+            "Waiting on block height {}",
+            chain_tip.metadata.stacks_block_height
+        );
+        let mut puppet_control = self.inner.lock().unwrap();
```

Comment:
Minor nit: instead of using an explicit `drop()`, could you put this into its own method?

Reply:
Added a new `with_lock` function to prevent these two `drop()`.